src/main/kotlin/com/nikitanazarov/coroutinestacks/CoroutineStacksPanel.kt

```diff
@@ -31,34 +37,319 @@ class CoroutineStacksPanel(project: Project) : JBPanelWithEmptyText() {
+
+        coroutineGraph?.add(coroutineStacksWindowHeader)
+
+        val coroutineStacksView = Box.createVerticalBox()
+        buildCoroutineStacksView(dispatcherToCoroutineStacksTree, coroutineStacksView, dispatchers)
+
+        coroutineGraph?.add(coroutineStacksView)
+
         add(coroutineGraph)
+}
+
+    private fun buildCoroutineStacksView(
+        mapOfParallelStackTree: MutableMap<String, Tree<CoroutineStacksNode>>,
+        coroutineStacksView: Box,
+        dispatchers: MutableSet<String>
+    ) {
+
+        if (dispatchers.isEmpty()) {
+            return
+        }
+
+        // code is incomplete, a coroutine stack view will show graphs for all possible dispatchers, but here
+        // the graph will be shown for only the first dispatcher in the list.
+        // Implement the functionality to build graph for all dispatchers. Fix this in next commit.
+
+        var dispatchersList = dispatchers.toTypedArray()
+        val tree = mapOfParallelStackTree[dispatchersList[0]] ?: return
+
+        val rows = mutableListOf<Box>()
+        for (i in 1..tree.getHeight()) {
+            rows.add(Box.createHorizontalBox())
+        }
+
+        tree.root?.let { addCoroutineInfoToCoroutineStackWindow(it, rows) }
+
+        for (i in rows.reversed()) {
+            coroutineStacksView?.add(i)
+            coroutineStacksView?.add(Box.createVerticalStrut(Constants.boxVerticalStruct))
+        }
+
+    }
+
+    private fun addCoroutineInfoToCoroutineStackWindow(
+        rootNode: TreeNode<CoroutineStacksNode>,
+        rows: MutableList<Box>
+    ) {
+        val stack = Stack<Pair<TreeNode<CoroutineStacksNode>, Int>>()
+        stack.push(rootNode to 0)
+
+        while (!stack.isEmpty()) {
+            val (node, level) = stack.pop()
+
+            if (node.value.stackTrace.isNotEmpty()) {
+                addCoroutineInfoBox(node.value, rows[level])
+            }
+
+            for (child in node.children) {
+                stack.push(child to level + 1)
+            }
+        }
     }
+
+    private fun addCoroutineInfoBox(
+        value: CoroutineStacksNode,
+        box: Box
+    ) {
+
+        val headerText = CoroutineStacksBundle.message("number.of.coroutines", value.additionalData.size)
+        val stackFrames = mutableListOf<String>()
+        stackFrames.add(headerText)
+
+        for (i in value.stackTrace) {
+            i?.let { stackFrames.add(it) }
+        }
+
+        val coroutineListView = JBList<String>(stackFrames)
+
+        if (cellRenderer != null)
+            coroutineListView.cellRenderer = cellRenderer
+
+        val scrollPane = JBScrollPane(coroutineListView)
+
+        val border = BorderFactory.createLineBorder(Color.BLACK, Constants.borderWidth)
+        scrollPane.border = border
+
+        box.add(scrollPane)
+        box.add(Box.createHorizontalStrut(Constants.boxHorizontalStruct))
+
+    }
+
+    private val cellRenderer = object : DefaultListCellRenderer() {
```

Comment:
Please, move the declaration of this variable to the top part of the class before methods are declared